service-base/src/main/java/org/eclipse/hono/service/command/CommandResponseSenderImpl.java

```diff
@@ -144,9 +143,7 @@ private static Message createResponseMessage(
         if (contentType != null) {
             msg.setContentType(contentType);
         }
-        if (payload != null) {
-            msg.setBody(new Data(new Binary(payload.getBytes())));
-        }
+        MessageHelper.setPayload(msg, contentType, payload);
```

Comment:
we can also remove the three lines above, can't we?

Reply:
You mean:

~~~java
if (contentType != null) {
  msg.setContentType(contentType);
}
~~~

I will re-test this. But IIRC then a some integration test failed. As this would mean that the content type is set. `setPayload` only sets the content type, if the body is not `null`.

Hono uses two approaches for the content type. Sometimes it is set when the body is `null`, sometimes it isn't.

We could always set this. Or add a parameter to the `setPayload` method and explicitly tell it how to handle this situation.

Reply:
So far, the policy is that we *always* set a content type on event and telemetry messages being forwarded downstream. There is only one case where we accept an empty body: the *empty-notification* as defined by the Event API. So, for the time being we should make sure that the content type gets set.

Reply:
Meaning, keep the existing logic for setting the content type.